geopandas/io/file.py

```diff
@@ -30,18 +32,27 @@ def _is_url(url):
         return False
 
 
-def read_file(filename, bbox=None, **kwargs):
+def read_file(filename, bbox=None, mask=None, row_filter=None, **kwargs):
     """
     Returns a GeoDataFrame from a file or URL.
 
+    .. versionadded:: 0.7.0 mask, row_filter
+
     Parameters
     ----------
     filename: str
         Either the absolute or relative path to the file or URL to
         be opened.
-    bbox : tuple | GeoDataFrame or GeoSeries, default None
-        Filter features by given bounding box, GeoSeries, or GeoDataFrame.
-        CRS mis-matches are resolved if given a GeoSeries or GeoDataFrame.
+    bbox: tuple | GeoDataFrame or GeoSeries | shapely Geometry, default None
+        Filter features by given bounding box, GeoSeries, GeoDataFrame or a
+        shapely geometry. CRS mis-matches are resolved if given a GeoSeries
+        or GeoDataFrame. Cannot be used with mask.
+    mask: dict | GeoDataFrame or GeoSeries | shapely Geometry, default None
+        Filter features by given dict-like geojson geometry, GeoSeries,
+        GeoDataFrame or a shapely geometry. CRS mis-matches are resolved
```

Comment:
Can you clarify *how* features are filtered? (you answered on that to a previous question of me about that with a quote from gdal, I think it uses intersects) 
As otherwis it is not very clear what the difference is with bbox

Reply:
Just updated. Thoughts?

Reply:
Looks good!